### DISCUSSION TITLE
scan's summary weird output

### DISCUSSION THREAD

**Discussion by daggs1** [endless-sky/endless-sky#7863] [Balance, Mechanics, and Interactions]
Greetings.

I'm working on adding "fog" to outfits which are unknown and I've stumbled on a bug which I cannot quite explain, here is a screenshot:
![es-scan-bad](https://user-images.githubusercontent.com/90052312/207171869-16909947-31df-47b6-b913-3247c564eba8.jpg)
as you can see, part of the unknown components has weird spacing between. I initially thought it is an issue with multiple unknown component of a category but I saw it happening when there is one unknown component.
printing the summery to stdout doesn't seems to reproduce the issue
code can be found at https://github.com/daggs1/endless-sky/tree/outfit_fog_of_war
ideas?

**Comment by vitalchip** [comment]
I think it's just formatting because your string is now over two lines, try knocking "component" off the end and see how it formats. I don't think it's needed anyway.

"2 Unknown Turrets" sounds better than "2 unknown Turrets Components"

const std::string PlayerInfo::ComposeUnDiscoveredOutfitName(const std::string category) const
{
	return 	"Unknown " + category + " Component";
}

**Reply by daggs1** [reply]
appending the word "Component" to the end makes lifes easier when it comes to plural categorizes.
I think that in the case of the scan, dropping the category is logical and will fix the issue.

Thanks for the suggestion

**Reply by daggs1** [reply]
removing it works but I have issues with the length of the boarding txt for secondary weapons

**Reply by vitalchip** [reply]
A lot of the panels are very small and restrictive when it comes to text length, probably find you're in for a lot of compromising to make stuff fit.

**Reply by daggs1** [reply]
understood that, I'll make the adjustments.
unrelated question, the outfitters location when pressing M is dynamic right? e.g. I assume it is populated when I find an outfitter?

**Reply by vitalchip** [reply]
Not sure what you mean.

When you press 'm' for map and then 'o' for outfitters it'll show the outfits you've seen and highlight those in the selected system which will be your current system if you've not selected any other.

**Reply by daggs1** [reply]
intriguing... it is possible to point me to the code bit?

**Reply by vitalchip** [reply]
void MapOutfitterPanel::DrawItems() and void MapOutfitterPanel::Init()  

Between them they create and display/or not outfits based on what the player has seen and where they are / are clicking on.

**Reply by daggs1** [reply]
weird, I looked there but didn't saw anything important
I'll look again

**Reply by daggs1** [reply]
now I see it, isn't this assumption weak? https://github.com/endless-sky/endless-sky/blob/02e7309936a0ae563a68bd303061f76462c2c540/source/MapOutfitterPanel.cpp#L266

the fact that I visited a planet doesn't means I've visited the outfitter...
maybe I should fix that too...
objections?

**Reply by vitalchip** [reply]
Probably more of an assumption than you were giving it credit for, you don't have to land on the planet to have 'seen' what the outfitter  sells, just have flown through the system.

I would think it's designed to be a generous abstraction to help fill the players map, particularly in the early game.
